include/geometry/polygon.h

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2023 UltiMaker
+// CuraEngine is released under the terms of the AGPLv3 or higher
+
+#ifndef GEOMETRY_POLYGON_H
+#define GEOMETRY_POLYGON_H
+
+#include "closed_polyline.h"
+
+namespace cura
+{
+
+class Shape;
+class ListPolyIt;
+class AngleDegrees;
+
+/*!
+ * \brief A Polygon is a specific type of polyline, for which we consider that the "inside" part of
+ *        the line forms a surface
+ * \sa https://github.com/Ultimaker/CuraEngine/wiki/Geometric-Base-Types#polygon
+ */
+class Polygon : public ClosedPolyline
```

Comment:
(A bit nitpicky, but:) This feels less like a class and more like a collection of methods that can be applied to `ClosedPolyline`s.

Reply:
Yeah that is true. One of my candidate architectures was to use only vector of points, and define static utility methods that work on them. But making classes adds a lot of explicity to the code IMHO. It also allows to define functions that can work only on specific types. For example, if you want to make a processing function that depends on the area, then you can only give a Polygon object to it, and not an OpenPolyline.